File: Design Patterns/Observer Pattern/CSharp/FaculdadeVincit.DesignPattern.ObserverPattern/Publisher.cs
using System.Security.Cryptography.X509Certificates;

namespace FaculdadeVincit.DesignPattern.ObserverPattern;

public abstract class Publisher<TValue>
{
    public List<IObservable<TValue>> Observables {get; private set;}

    public Publisher()
    {
        Observables = new List<IObservable<TValue>>();

    }

    public void AddObservable(IObservable<TValue> observable) => Observables.Add(observable);

    public void RemoveObservable(IObservable<TValue> observable) => Observables.Remove(observable);

    protected void NotifyObservables(TValue valor) => Observables.ForEach(o => o.Update(valor));
}

File: Design Patterns/Observer Pattern/Java/com.faculdadevincit.designpatterns.observerpattern/src/Publisher.java
import java.util.ArrayList;
import java.util.List;

public class Publisher<TValue> {
    private List<IObservable<TValue>> observables;

    protected List<IObservable<TValue>> getObservables() {
        return observables;
    }

    public Publisher() {
        observables = new ArrayList<>();
    }

    public void addObservable(IObservable<TValue> observable) {
        observables.add(observable);
    }

    public void removeObservable(IObservable<TValue> observable) {
        observables.remove(observable);
    }

    public void notifyObservables(TValue value) {
        observables.forEach(o -> o.update(value));
    }
}
